src/server/pfs/server/driver_v2.go

```diff
@@ -173,6 +175,28 @@ func (d *driverV2) withFileSet(ctx context.Context, repo, commit string, cb func
 	})
 }
 
+func (d *driverV2) withWriter(ctx context.Context, commit *pfs.Commit, cb func(int64, *fileset.Writer) error) (retErr error) {
+	n := d.getSubFileSet()
+	subFileSetStr := fileset.SubFileSetStr(n)
+	subFileSetPath := path.Join(commit.Repo.Name, commit.ID, subFileSetStr)
+	fsw := d.storage.NewWriter(ctx, path.Join(tmpPrefix, subFileSetPath))
+	defer func() {
+		if err := d.storage.Delete(ctx, path.Join(tmpPrefix, subFileSetPath)); retErr == nil {
+			retErr = err
+		}
+	}()
+	if err := cb(n, fsw); err != nil {
+		return err
+	}
+	if err := fsw.Close(); err != nil {
+		return err
+	}
+	return d.compactionQueue.RunTaskBlock(ctx, func(m *work.Master) error {
```

Comment:
I don't think this needs to be compacted. I think you could just do a copy object. Also, we should probably add special casing for a compaction that only involves one subfileset to just copy the object.

Reply:
I changed this to a call to `d.storage.Compact` which we can change later to handle that case well.